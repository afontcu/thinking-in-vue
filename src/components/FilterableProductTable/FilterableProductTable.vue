<template>
  <div class="filterable-product-table">
    <SearchBar />
    <ProductTable :products="products"/>
  </div>
</template>

<script>
import SearchBar from '@/components/SearchBar'
import ProductTable from '@/components/ProductTable'

export default {
  name: 'FilterableProductTable',
  components: {
    SearchBar,
    ProductTable
  },
  props: {
    products: {
      type: Array,
      required: true
    }
  },
  data () {
    return {}
  }
}
</script>

<style>
.filterable-product-table {
  height: auto;
  width: 600px;
  padding: 1.5rem;
  border-radius: 8px;
  background-color: white;
  box-shadow: 0 2px 6px hsla(0, 0%, 0%, 0.2)
}
</style>
