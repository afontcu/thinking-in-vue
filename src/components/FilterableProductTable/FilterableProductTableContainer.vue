<template>
  <FilterableProductTable
    :products="products"
    :filterText="filterText"
    :displayOnlyOffers="displayOnlyOffers" />
</template>

<script>
import FilterableProductTable from './FilterableProductTable'
import products from '@/api.json'

export default {
  name: 'FilterableProductTableContainer',
  components: { FilterableProductTable },
  data () {
    return {
      products,
      filterText: '',
      displayOnlyOffers: false
    }
  }
}
</script>
