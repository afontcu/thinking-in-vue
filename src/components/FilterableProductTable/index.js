import FilterableProductTable from './FilterableProductTable'

export default FilterableProductTable
