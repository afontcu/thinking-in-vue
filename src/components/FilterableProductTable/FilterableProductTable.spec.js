import FilterableProductTable from './index.js'
import { mount } from '@vue/test-utils'

const propsData = {
  products: [
    {
      'id': 4450,
      'name': 'Producte 01',
      'offer': true,
      'price': 14.61
    },
    {
      'id': 2212,
      'name': 'Producte 02',
      'offer': true,
      'price': 21.02
    }]
}

describe('FilterableProductTable.vue', () => {
  it('is a component', () => {
    const wrapper = mount(FilterableProductTable, { propsData })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(FilterableProductTable, { propsData })
    expect(wrapper.classes()).toContain('filterable-product-table')
  })

  it('renders an snapshot', () => {
    const wrapper = mount(FilterableProductTable, { propsData })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
