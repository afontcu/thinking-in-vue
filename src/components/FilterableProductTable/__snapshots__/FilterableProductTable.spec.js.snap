// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableProductTable.vue renders an snapshot: default 1`] = `
<div class="filterable-product-table">
    <form>
(??)
        <p class="filterOffers">
            <input type="checkbox" id="filterOffers">
            <label for="filterOffers">Show only offers</label>
        </p>
    </form>
    <table>
        <thead>
            <tr>
                <th>Name</th>
                <th>Price</th>
            </tr>
        </thead>
        <tbody>
(??)
                </td>
(??)
            </tr>
(??)
                </td>
(??)
            </tr>
        </tbody>
    </table>
</div>
`;
